### PR TITLE
Set up ZipCodeSummaryTable servlet

### PR DESCRIPTION
**Pull request by jennygaoo**
WIP

### CODE REVIEW

File: postal-injection/src/main/java/com/google/sps/servlets/ZipCodeSummaryTableServlet.java [hunk 1]
```diff
@@ -21,12 +20,16 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // TODO: add parameters for GetZipCodeSummary.getZipCodeSummary();
     String zipCode = request.getParameter("zip-code");
     float regularShippingRate = request.getParameter("regular-rate");
     float palletShippingRate = request.getParameter("pallet-rate");
+    int minimumPalletSize = request.getParameter("min-pallet-size");
+    int threshold = request.getparameter("threshold");
+    float postOfficeToCustomerRate = request.getParameter("PO-to-customer-rate");
```
Comment (jennygaoo) [inline postal-injection/src/main/java/com/google/sps/servlets/ZipCodeSummaryTableServlet.java]: wondering if these condensed parameter names, like "regular-rate", "pallet-rate", etc. are okay. I don't want to make the URL param too long

Comment (andreahu12) [reply postal-injection/src/main/java/com/google/sps/servlets/ZipCodeSummaryTableServlet.java]: regular-rate and pallet-rate lgtm

File: postal-injection/src/test/java/com/google/sps/OrderDestinationMapServletTest.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      new HashMap<String, ZipCodeData>() {
+        {
+          put("12123", ZIPCODE_A);
+          put("21121", ZIPCODE_B);
+        }
+      };
+
+  private static final float EXPECTED_ZIPCODE_A_ORDER_DENSITY = 1.0f;
+  private static final float EXPECTED_ZIPCODE_B_ORDER_DENSITY = 2.0f / 3.0f;
+  private static final float FLOATING_POINT_DELTA = 0.00001f;
```
Comment (andreahu12) [inline postal-injection/src/test/java/com/google/sps/OrderDestinationMapServletTest.java:56]: Move all constants to the top of the file

File: postal-injection/src/main/java/com/google/sps/servlets/OrderDestinationMapServlet.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    HashMap<String, ZipCodeData> zipcodeData = ZipCodeDataProcessor.getZipcodeData();
+    ArrayList<ZipcodeSummary> listOfData = summarizeData(zipcodeData);
+    String json = gson.toJson(listOfData);
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  public ArrayList<ZipcodeSummary> summarizeData(HashMap<String, ZipCodeData> zipcodeData) {
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/OrderDestinationMapServlet.java:41]: private? and can this be static?

File: postal-injection/src/main/java/com/google/sps/servlets/OrderDestinationMapServlet.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+      }
+    }
+    for (String key : zipcodeData.keySet()) {
+      listOfData.add(new ZipcodeSummary(zipcodeData.get(key), highestOrderCount));
+    }
+    return listOfData;
+  }
+
+  class ZipcodeSummary {
+    private final String zipcode;
```
Comment (jgao23) [inline postal-injection/src/main/java/com/google/sps/servlets/OrderDestinationMapServlet.java:58]: supernit: zipCode? let's try be consistent about the capitalization of zip code throughout the code

**Comment by jennygaoo** [comment]
> Nice! What about unit tests?

I was thinking about writing unit tests, however, the servlet pretty much just calls

**Pull request merged**